### PR TITLE
Improve user id validation in oauth

### LINKED ISSUE #1705: Validate `user_id` in OAuth views
#### Summary

We need to do better frontend validation of the user ID field in OAuth views.

#### Why do we need this?

We're seeing that users are making too many mistakes with user IDs in the OAuth views

#### What is already there? What do you see now?

First of all, the validations are inconsistent. We sometimes see

```js
user_id: Yup.string().required(sharedMessages.validateRequired)
```

while in other places we also see a min/max length.

#### What is missing? What do you want to see?

The user_id should be validated the same way everywhere:

- Required
- Min length 3
- Max length 36
- Regex `^[a-z0-9](?:[-]?[a-z0-9]){2,}$`

> NOTE: maybe out of scope for this issue, but it might be good to check all ID fields in the console as well. All IDs have the same rules.

#### How do you propose to implement this?

Write once, copy-paste many.

### PR DESCRIPTION
**Pull request by pgalic96**
#### Summary

Closes #1705 

#### Changes

- Add user regex
- Improve user validation and add it to all `oauth` views

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/lib/regexp.js [hunk 1]
```diff
@@ -15,3 +15,4 @@
 /* eslint-disable import/prefer-default-export */
 
 export const url = /[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)?/gi
+export const user = /^[a-z0-9](?:[-]?[a-z0-9]){2,}$/
```
Comment (htdvisser) [inline pkg/webui/lib/regexp.js]: I'd just call this `id` as it's the same for all IDs.

File: pkg/webui/lib/shared-messages.js [hunk 2]
```diff
@@ -215,6 +215,7 @@ export default defineMessages({
   validateTooLong: 'Too long',
   validateTooShort: 'Too short',
   validateUrl: 'Invalid URL format',
+  validateUserFormat: 'Username can contain only dash (-) as the special char',
```
Comment (htdvisser) [inline pkg/webui/lib/shared-messages.js]: (any) ID can be 3-36 characters (already covered by your other validations) and can only contain lowercase letters, numbers and dashes (-)

Comment (pgalic96) [reply pkg/webui/lib/shared-messages.js]: alright. there are some cases not covered though, but in my opinion its too specific to be mentioned. these cases are if the username is for example `p-g` or any id that has dash in the end `tti-` for example.

Comment (htdvisser) [reply pkg/webui/lib/shared-messages.js]: I agree. If someone tries to use `----` as a username, they can probably already imagine that we wouldn't accept it.

File: pkg/webui/lib/regexp.js [hunk 3]
```diff
@@ -15,3 +15,4 @@
 /* eslint-disable import/prefer-default-export */
```
Comment (kschiffer) [inline pkg/webui/lib/regexp.js]:
```suggestion

```

File: pkg/webui/lib/shared-messages.js [hunk 4]
```diff
@@ -215,6 +215,7 @@ export default defineMessages({
   validateTooLong: 'Too long',
   validateTooShort: 'Too short',
   validateUrl: 'Invalid URL format',
+  validateIdFormat: 'ID can only contain lowercase letters, numbers and dashes (-)',
```
Comment (kschiffer) [inline pkg/webui/lib/shared-messages.js]:
```suggestion
  validateIdFormat: 'IDs can only contain lowercase letters, numbers and dashes (-)',
```

**Pull request merged**